crates/flowctl/src/source/unbundle.rs

```diff
@@ -0,0 +1,338 @@
+use crate::source::Existing;
+use anyhow::Context;
+use serde_json::value::RawValue;
+use serde_json::Value;
+use std::convert::AsRef;
+use std::path::Path;
+use tokio::io::AsyncWriteExt;
+use tokio::{fs, io::AsyncReadExt};
+
+/// Unbundle is the opposite of `bundle`. It takes a catalog and writes it out
+/// as file(s) in the given `dir_path`. The input catalog is typically one that
+/// is bundled, but this is not a requirement. If the given catalog contains
+/// inlined resources, then they will be written out as separate files and the
+/// spec will have them replaced with relative paths to them. Resources that
+/// are _not_ inlined will be left unmodified. Note that `unbundle` does _not_
+/// put things into nested subdirectories. The entire catalog will be written
+/// directly within the given directory.
+pub async fn unbundle(
+    new_catalog: models::Catalog,
+    dir_path: &Path,
+    spec_filename: &str,
+    json: bool,
+    existing: Existing,
+) -> anyhow::Result<()> {
+    fs::create_dir_all(dir_path).await?;
+    let flow_yaml_path = dir_path.join(spec_filename);
+
+    let maybe_catalog = resolve_catalog_to_write(&flow_yaml_path, new_catalog, existing).await?;
+    let Some(mut new_catalog) = maybe_catalog else {
+        return Ok(());
```

Comment:
Silently do nothing if we don't pass a valid flow catalog to unbundle? Should this error instead?

Reply:
The error is being propagated here (`?` at the end of line 28). But the successful result of `resolve_catalog_to_write` might still be `None` if the user has opted not to overwrite existing catalog specs with the new one, and there's an existing file already there.

Reply:
Gotcha!